assembl/models/auth.py

```diff
@@ -1752,7 +1752,8 @@ def getCurrent(cls, req=None, session=None):
             user_id = req.authenticated_userid
             if user_id and user_id != Everyone:
                 try:
-                    discussion_id = req.matchdict['discussion_id'] if (req.matchdict and 'discussion_id' in req.matchdict) else None
+                    discussion_id = req.matchdict['discussion_id'] if (
+                        'matchdict' in req.__dict__ and req.matchdict and 'discussion_id' in req.matchdict) else None
```

Comment:
This feels hacky and complicated. matchdict really can be missing in req?
I tend to prefer `.get` and `gettattr` syntax instead of `in` like you did. This line should do the same:

    discussion_id = req.matchdict.get('discussion_id', None) if getattr(req, 'matchdict', None) is not None else None

If we always have a req.matchdict attribute, an empty dict or None, even simpler:

    discussion_id = req.matchdict.get('discussion_id', None)